Improve CI for Linux

andi-makes opened the pull request:
Changing the build command for the linux-release job to use [cross](https://github.com/cross-rs/cross). The linux target gets compiled against `glibc` version `2.17`, the lowest `rust` >= `1.60` itself supports, which should maximize compatibility for many linux distros.

If desired, we can also use cross to setup a build using `musl` instead of `libc`. According to my testing, a binary built with `musl` doesn't work on wayland though, because of [this issue](https://github.com/rust-windowing/winit/issues/1818).

I also implemented a simple `cargo check` workflow for every platform, which is currently triggered by a push.

At the moment, this is an initial draft. Things I would like to improve further are:
 - [x] Caching
 - [x] Run `cargo clippy` on push
 - [x] Run `cargo fmt --check` on push

If this PR looks like it is going in the right direction, I'd like to continue working on above points, otherwise I'm open to change the PR according to your needs.

andi-makes committed:
Use Cross for CI compilation

andi-makes committed:
Run cargo check on push

andi-makes committed:
[skip ci] Fix job names

andi-makes committed:
Maybe added working cache

andi-makes committed:
Changed cargo check to clippy, added fmt check

andi-makes committed:
Made clippy fail on warnings

andi-makes committed:
Now for real, clippy fails on warnings

andi-makes committed:
Fixed release type and path to executable

andi-makes committed:
Fix cache for linux release

CheaterCodes commented:
From what I understand, you're targeting `main-centos`, which seems to me like a moving tag. If we don't want unexpected surprises, maybe we should use something like `0.2.4-centos`?
I'm not sure if I understood this correctly, and I don't know which image has which glibc version, but I'm worried that the current setup will stop working one day.

andi-makes committed:
Added comments to Cross.toml, pinned 0.2.4-centos

Added some comments to the Cross.toml file to provide further
information of the glibc version that will be used corresponding with
the build image.

Additionally, pinned the build image version to 0.2.4-centos.

andi-makes commented:
Good point on the build image version, changed that in the most recent commit.

I also added some comments which are providing information to the corresponding glibc version and where to find that information.

andi-makes commented:
It's been half a year, and I am no longer interested in this PR. Feel free to use this PR as a starting point if desired.
